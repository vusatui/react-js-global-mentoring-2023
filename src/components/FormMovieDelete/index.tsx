import {
    StyledFormMovieDeleteFooter,
    StyledFormMovieDeleteSubmitButton,
    StyledFormMovieDeleteTitle,
    StyledFormMovieDeleteWarning,
    StyledFormMovieDeleteWrapper
} from "./styled";

interface FormMovieDeleteProps {
    className?: string
    onConfirm: () => void
}

const FormMovieDelete = ({
    className,
    onConfirm,
}: FormMovieDeleteProps) => (
    <StyledFormMovieDeleteWrapper className={className}>
        <StyledFormMovieDeleteTitle>
            DELETE MOVIE
        </StyledFormMovieDeleteTitle>
        <StyledFormMovieDeleteWarning>
            Are you sure you want to delete this movie?
        </StyledFormMovieDeleteWarning>
        <StyledFormMovieDeleteFooter>
            <StyledFormMovieDeleteSubmitButton onClick={onConfirm}>
                CONFIRM
            </StyledFormMovieDeleteSubmitButton>
        </StyledFormMovieDeleteFooter>
    </StyledFormMovieDeleteWrapper>
);

export default FormMovieDelete;