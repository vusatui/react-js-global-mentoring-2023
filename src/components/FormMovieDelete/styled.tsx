import styled from "styled-components";

export const StyledFormMovieDeleteWrapper = styled.div`
    width: 600px;
`;

export const StyledFormMovieDeleteTitle = styled.div`
  margin-bottom: 37px;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 40px;
  line-height: 49px;
  letter-spacing: 1px;
  text-transform: uppercase;
  color: #FFFFFF;

`;

export const StyledFormMovieDeleteWarning = styled.div`
  margin-bottom: 52px;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 400;
  font-size: 20px;
  line-height: 24px;
  color: #FFFFFF;
`;

export const StyledFormMovieDeleteFooter = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-end;
  align-items: center;
`;

export const StyledFormMovieDeleteSubmitButton = styled.button`
  cursor: pointer;
  border: none;
  outline: none;
  appearance: none;
  padding: 17px 36px;
  background: #F65261;
  border-radius: 4px;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: 24px;
  text-align: center;
  text-transform: uppercase;
  color: #FFFFFF;
`;