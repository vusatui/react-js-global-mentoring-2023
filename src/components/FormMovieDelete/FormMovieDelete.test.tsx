import FormMovieDelete from "./index";
import {render} from "@testing-library/react";

const handleConfirm = jest.fn();

const Component = () => (<FormMovieDelete  onConfirm={handleConfirm}/>)

describe("FormMovieDelete", () => {

    it("Should renders correctly", () => {
        const { asFragment } = render(<Component />);
        expect(asFragment()).toMatchSnapshot();
    });

});