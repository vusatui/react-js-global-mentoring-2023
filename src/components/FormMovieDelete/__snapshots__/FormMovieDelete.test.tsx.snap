// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMovieDelete Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB hNNRpq"
  >
    <div
      class="sc-gueYoa ijIEeo"
    >
      DELETE MOVIE
    </div>
    <div
      class="sc-dmqHEX jVOWGX"
    >
      Are you sure you want to delete this movie?
    </div>
    <div
      class="sc-hLseeU cOqkZB"
    >
      <button
        class="sc-eDDNvR bffkXQ"
      >
        CONFIRM
      </button>
    </div>
  </div>
</DocumentFragment>
`;
